File: src/Hurace/Hurace.Core.Dto/Skier.cs
using System;
using System.ComponentModel.DataAnnotations;
using Hurace.Core.Dto.Attributes;
using Hurace.Core.Dto.Interfaces;

namespace Hurace.Core.Dto
{
    public class Skier : ISinglePkEntity
    {
        [Key]
        public int Id { get; set; }

        public string FirstName { get; set; } = default!;
        public string LastName { get; set; } = default!;
        public DateTime DateOfBirth { get; set; }
        public int CountryId { get; set; }
        [Navigational]
        public Country? Country { get; set; }
        public int GenderId { get; set; }
        [Navigational]
        public Gender? Gender { get; set; }
    }
}

File: src/hurace-web-ts/src/interfaces/Skier.ts
import { Country } from './Country';
import { Gender } from './Gender';

export interface Skier {
    id: number;
    firstName: string;
    lastName: string;
    dateOfBirth: Date;
    countryId: number;
    country: Country;
    genderId: number;
    gender: Gender;
    imageUrl?: string;
}
